app/assets/javascripts/reducers/users.js

```diff
@@ -23,12 +23,34 @@ const SORT_DESCENDING = {
 
 export default function users(state = initialState, action) {
   switch (action.type) {
-    case RECEIVE_USERS: return {
+    case RECEIVE_USERS: {
+      // Get the sorting key if available from Redux store or else use last_name
+      let sort_key = state.sort.key || 'last_name';
+      let previousKey = null;
+      // Initialize a variable to hold the user list.
+      let user_list = action.data.course.users;
+
+       // Check if any user in the user_list has 'real_name'
+       if (user_list.some(user => user.real_name)) {
+         // If any users have 'real_name', transform the 'real_name' into separate
+         // 'first_name' and 'last_name' properties and update the user list
+         user_list = transformUsers(user_list);
+        } else if (!state.sort.key) {
+          // If there are no users with real_name and key in the store is null then set sort_key by username
+          sort_key = 'username';
+        }
+
+        if (state.sort.key) { previousKey = state.sort.sortKey ? null : sort_key; }
```

Comment:
can you add a comment here explaining why/how this works?

Reply:
Okay sure